packages/core/state/interaction/logics.ts

```diff
@@ -69,12 +69,22 @@ const initializeApp = createLogic({
         // If there are query args representing a query we can extract that
         // into the query to render (ex. when refreshing a page)
         if (isOnWeb && window.location.search) {
-            dispatch(
-                selection.actions.addQuery({
-                    name: DEFAULT_QUERY_NAME,
-                    parts: SearchParams.decode(window.location.search),
-                })
-            );
+            try {
+                dispatch(
+                    selection.actions.addQuery({
+                        name: DEFAULT_QUERY_NAME,
+                        parts: SearchParams.decode(window.location.search),
+                    })
+                );
+            } catch (err) {
+                // Parsing error from SearchParams.decode
```

Comment:
This catches all errors correct? does this comment align with that?

Reply:
Technically no, it only catches the decode error since `addQuery` has its own error handling. I could write this equivalently (which may have been more clear) if I separated `const parts = SearchParams.decode()` as its own step and just wrapped that, but then I'd need to separately prevent `addQuery` from running...